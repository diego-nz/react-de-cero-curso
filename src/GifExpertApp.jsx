import React, { useState } from 'react';
import { CategorieAdd } from './components/CategorieAdd';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
    
    const [categories, setCategories] = useState(['']);
    // const [inputValue, setInputValue] = useState('')

    // const handleAdd = () => {
    // Para el control del input
        
    // ** Otra forma de hacer Callback que devuelve el estado anterio + lo nuevo
    // setCategories( cats => [...cats,'hiigi','hooal']);

    // setCategories([...categories, inputValue]);
    // setInputValue('');
    // }

    // Para settear valor al input
    // const handleInput = (e) => {
    //     setInputValue(e.target.value);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />
            
            {/* El input */}
            {/* <input type="text" onChange={(e) => handleInput(e)} value={inputValue}/> */}
            {/* <button value="Agregar" onClick={handleAdd}>Agregar</button> */}

            <CategorieAdd setCategories={ setCategories } />

            {/* <ul> */}
                {categories.map((value, index) => {
                    // return <li key={index}>{value}</li>
                    // Si no quiero poner return, quitar las llaves
                    return <GifGrid 
                        categorie={value} 
                        key={index} 
                    />
                })}
            {/* </ul> */}
        </>
    )
}
