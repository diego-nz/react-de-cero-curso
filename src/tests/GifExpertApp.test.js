import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";


describe('Pruebas del componente GifExpertApp', () => {
    it('debe mostrar correctamente el componente', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });
    it('debe de mostrar una lista de categorias', () => {
        const categories = ['Dragon ball', 'Naruto'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>)
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    });
});