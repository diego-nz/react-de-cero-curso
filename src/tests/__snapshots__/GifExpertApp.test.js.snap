// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <CategorieAdd
    setCategories={[Function]}
  />
  <GifGrid
    categorie="Dragon ball"
    key="0"
  />
  <GifGrid
    categorie="Naruto"
    key="1"
  />
</Fragment>
`;

exports[`Pruebas del componente GifExpertApp debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <CategorieAdd
    setCategories={[Function]}
  />
</Fragment>
`;
